mat64/dense.go

```diff
@@ -417,3 +420,25 @@ func (m *Dense) Augment(a, b Matrix) {
 	w := m.View(0, ac, br, bc).(*Dense)
 	w.Copy(b)
 }
+
+// MarshalBinary encodes the receiver into a binary form and returns the result.
+func (m Dense) MarshalBinary() ([]byte, error) {
+	buf := new(bytes.Buffer)
+	enc := binary.NewEncoder(buf)
```

Comment:
@btracey my rational for using an `encoding/binary` (or `gonuts/binary`) based solution is cross-language support.
`encoding/gob.GobEncoding` (and its `GobDecoding` sidekick) is obviously `go` oriented.

`varint` is rather easily implementable in various language (even `python`)

but I do recognize the usefulness of concurrent decoding capabilities that not using varint (and thus `int64`) buys us.